app/lib/notifications/videoConf/backgroundNotificationHandler.ts

```diff
@@ -0,0 +1,94 @@
+import notifee, { AndroidCategory, AndroidImportance, AndroidVisibility } from '@notifee/react-native';
+import messaging from '@react-native-firebase/messaging';
+import ejson from 'ejson';
+
+import { deepLinkingClickCallPush } from '../../../actions/deepLinking';
+import i18n from '../../../i18n';
+import { BACKGROUND_PUSH_COLOR } from '../../constants';
+import { store } from '../../store/auxStore';
+
+const backgroundNotificationHandler = async (): Promise<void> => {
+	await notifee.createChannel({
+		id: 'video-conf-call',
+		name: 'Video Call',
+		lights: true,
+		vibration: true,
+		importance: AndroidImportance.HIGH,
+		sound: 'ringtone'
+	});
+
+	notifee.onBackgroundEvent(async event => {
+		if (event.detail.pressAction?.id === 'accept' || event.detail.pressAction?.id === 'decline') {
+			const notificationData = event.detail?.notification?.data;
+			if (typeof notificationData?.caller === 'object' && (notificationData.caller as any)._id) {
+				store.dispatch(deepLinkingClickCallPush({ ...notificationData, event: event.detail.pressAction?.id }));
+				await notifee.cancelNotification(
+					`${notificationData.rid}${(notificationData.caller as any)._id}`.replace(/[^A-Za-z0-9]/g, '')
+				);
+			}
+		}
+	});
+};
+
+interface NotificationData {
+	notificationType?: string;
```

Comment:
You could do a discriminated union here and get rid of all the optionals.
You already do a `if (notification?.notificationType === 'videoconf') {`, so when that's the case, everything is required.